### PR TITLE
Add NOTICE file for Beats

### PR DESCRIPTION
**Pull request by monicasarbu**
This PR contains a python script that generates the NOTICE file. There is a single NOTICE file for all the official Beats, and it's placed in the [beats repository](https://github.com/elastic/beats).

The NOTICE file includes the copyright information for all the 3rd party libraries used in the Beats project. It follows the rules:
- if the library has apache license, and if NOTICE file is present, include that text in the NOTICE file. 
- if the library has MIT, ISC or BSD license, include the whole copyright block in NOTICE file (meaning copyright + license)

You can pass a list of vendor directories where the NOTICE* and LICENSE* files of all 3rd party libraries are available:
```
python dev-tools/generate_notice.py vendor metricbeat/vendor
```

For the official Beats, you can run the following in the [beats repository](https://github.com/elastic/beats).

```
make notice
```

The NOTICE file can be generated by the community Beats by using the script. The NOTICE file is not added by the Beat generator.

UPDATE: The script accepts also the name of the Beat and the copyright owner, to customize the NOTICE file for the community Beats.

```
$ python dev-tools/generate_notice.py -h
usage: generate_notice.py [-h] [-b BEAT] [-c COPYRIGHT] vendors [vendors ...]

Generate the NOTICE file from a list of vendor directories

positional arguments:
  vendors               List of vendor directories

optional arguments:
  -h, --help            show this help message and exit
  -b BEAT, --beat BEAT  Beat name
  -c COPYRIGHT, --copyright COPYRIGHT
                        Copyright owner
```

### CODE REVIEW

File: metricbeat/NOTICE [hunk 1]
```diff
@@ -0,0 +1,1692 @@
+License /Users/monica/src/github.com/elastic/beats/metricbeat/vendor/github.com/mitchellh/hashstructure/LICENSE
```
Comment (ruflin) [inline metricbeat/NOTICE]: We should not include here the full path as it contains your private username.

File: dev-tools/generate_notice.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+if __name__ == "__main__":
+
+	beat_dir = os.getcwd()
+	libbeat_dir = os.path.join(beat_dir, "../")
+	notice = os.path.join(beat_dir, "NOTICE")
+	vendors = [beat_dir, libbeat_dir]
+	licenses = []
+
+	for vendor in vendors:
```
Comment (ruflin) [inline dev-tools/generate_notice.py]: Will this also include vendor directories inside metricbeat modules?

File: metricbeat/NOTICE (hunk 1 above)
Comment (monicasarbu) [inline metricbeat/NOTICE]: Yes, sure. The plan is to include only `github.com/mitchellh/hashstructure`.

**Comment by ruflin** [comment]
This could also be interesting to see which repos do not contain a license file: https://github.com/pmezard/licenses

**Comment by cyrilleverrier** [comment]
This PR is going to be useful for community beats as well :)

Could you please make sure that we (developers) can configure the output by using variables instead of `Elastic Beat` and `Copyright 2014-{} Elasticsearch BV` ?

**Comment by monicasarbu** [comment]
@cyrilleverrier Thank you for such a quick feedback. I added two options (`-b` and `-c`) to configure the Beat name and the copyright owner. I have updated the description of the PR with the script usage.

File: Makefile [hunk 3]
```diff
@@ -113,3 +113,9 @@ upload-package:
 .PHONY: release-upload
 upload-release:
 	aws s3 cp --recursive --acl public-read build/upload s3://download.elasticsearch.org/beats/
+
+.PHONY: notice
+notice: 
+	python dev-tools/generate_notice.py metricbeat/vendor vendor
```
Comment (ruflin) [inline Makefile]: This does not include the vendor directories in the metricbeat modules? How about somthing like `find . -name vendor`?

File: dev-tools/generate_notice.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        file_content = f.read()
+        return file_content
+
+
+def get_library_name(license):
+
+    lib = ""
+    path = os.path.dirname(license)
+    # get the last three directories
+    for i in range(0, 3):
```
Comment (ruflin) [inline dev-tools/generate_notice.py]: Does this also work for dependencies not on Github?

File: Makefile (hunk 3 above)
Comment (monicasarbu) [inline Makefile]: Ah, good point. Then I will search in a given director for all the vendor directories.

File: dev-tools/generate_notice.py (hunk 4 above)
Comment (monicasarbu) [inline dev-tools/generate_notice.py]: It gets only the last 3rd directories from the path, but the function should be able to get less than 3 directories as well. I adjusted the code a bit to accept also less than 3 directories (example windows/project)

**Pull request merged**